### PR TITLE
edge: throw some SLD/SRD state errors

### PR DESCRIPTION
**Pull request by fippo**
as described in https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-18#section-5.5
and https://tools.ietf.org/html/draft-ietf-rtcweb-jsep-18#section-5.6

(merging at next minor release is sufficient)

### CODE REVIEW

File: src/js/edge/rtcpeerconnection_shim.js [hunk 1]
```diff
@@ -478,6 +478,22 @@ module.exports = function(edgeVersion) {
 
   RTCPeerConnection.prototype.setLocalDescription = function(description) {
     var self = this;
+
+    if ((description.type === 'offer' &&
```
Comment (KaptenJansson) [inline src/js/edge/rtcpeerconnection_shim.js]: For the sake of readability I would say the number of checks are on the limit, WDYT about putting them in a function with a descriptive name that returns a bool?

File: src/js/edge/rtcpeerconnection_shim.js [hunk 2]
```diff
@@ -588,6 +604,22 @@ module.exports = function(edgeVersion) {
 
   RTCPeerConnection.prototype.setRemoteDescription = function(description) {
     var self = this;
+
+    if ((description.type === 'offer' &&
```
Comment (KaptenJansson) [inline src/js/edge/rtcpeerconnection_shim.js]: Same as previous comment

File: src/js/edge/rtcpeerconnection_shim.js (hunk 1 above)
Comment (fippo) [reply src/js/edge/rtcpeerconnection_shim.js]: hrm. essentially that should be a function that checks if the state transition is allowed. I like that idea!

Comment (fippo) [reply src/js/edge/rtcpeerconnection_shim.js]: done

**Comment by fippo** [comment]
@MirkoBonadei can you take a look at this on too? (give me an inch... :-))

File: src/js/edge/rtcpeerconnection_shim.js [hunk 3]
```diff
@@ -478,6 +491,18 @@ module.exports = function(edgeVersion) {
 
   RTCPeerConnection.prototype.setLocalDescription = function(description) {
     var self = this;
+
+    if (!isActionAllowedInSignalingState('setLocalDescription',
+        description.type, this.signalingState)) {
+      var e = new Error('Can not set remote ' + description.type +
```
Comment (MirkoBonadei) [inline src/js/edge/rtcpeerconnection_shim.js]: I guess the error message should start with: 'Can not set local ', right?

**Review by MirkoBonadei — CHANGES_REQUESTED** [review]
lgtm, just a question about the error message.

File: src/js/edge/rtcpeerconnection_shim.js (hunk 3 above)
Comment (fippo) [reply src/js/edge/rtcpeerconnection_shim.js]: 🦅 👁 

fixed + squashed!

**Pull request merged**